src/core/file_format/trp.cpp

```diff
@@ -0,0 +1,142 @@
+#include <unistd.h>
+#endif
+
+TRP::TRP() = default;
+TRP::~TRP() = default;
+
+void TRP::GetNPcommID(std::filesystem::path trophyPath, int fileNbr) {
+    std::filesystem::path trpPath = trophyPath / "sce_sys/npbind.dat";
+    Common::FS::IOFile npbindFile(trpPath, Common::FS::FileAccessMode::Read);
+    if (!npbindFile.IsOpen()) {
+        return;
+    }
+    for (int i = 0; i < fileNbr; i++) {
+        std::vector<u8> vec(12);
+        npbindFile.Seek(0x84 + (i * 0x180));
+        npbindFile.Read(vec);
+        vec.resize(16);
+        NPcommID.push_back(vec);
+    }
+    npbindFile.Close();
+}
+
+void TRP::removePadding(std::vector<u8>& vec) {
+    for (auto it = vec.rbegin(); it != vec.rend(); ++it) {
+        if (*it == '>') {
+            size_t pos = std::distance(vec.begin(), it.base());
+            vec.resize(pos);
+            break;
+        }
+    }
+}
+
+bool TRP::Extract(std::filesystem::path trophyPath) {
+    std::string title = trophyPath.filename().string();
+    std::filesystem::path gameSysDir = trophyPath / "sce_sys/trophy/";
+    if (!std::filesystem::exists(gameSysDir)) {
+        return false;
+    }
+    std::vector<std::filesystem::path> fileList;
+    for (const auto& entry : std::filesystem::directory_iterator(gameSysDir)) {
+        if (entry.is_regular_file()) {
+            fileList.push_back(entry.path());
+        }
+    }
+
+    if (fileList.empty())
+        return false;
+
+    GetNPcommID(trophyPath, fileList.size());
+    for (int nbr = 0; const std::filesystem::path& fileinfo : fileList) {
+        std::string multiTrp = fileinfo.stem().string();
+
+        Common::FS::IOFile file(fileinfo, Common::FS::FileAccessMode::Read);
+        if (!file.IsOpen()) {
+            return false;
+        }
+
+        trp_header header;
+        file.ReadRaw<u8>(&header, sizeof(trp_header));
+
+        if (header.magic != 0xDCA24D00)
+            return false;
+
+        s64 seekPos = sizeof(trp_header);
+
+        char buf[1024];
+
+#ifdef _WIN64
+        GetModuleFileNameA(NULL, buf, 1024);
+#elif
+        if (getcwd(buffer, sizeof(buf)) == NULL) {
+            // std::cerr << "Error getting current directory!" << std::endl;
+            return false;
+        }
+#endif
+        std::string appPath(buf);
+        std::size_t found = appPath.find_last_of("/\\");
+        if (found != std::string::npos) {
+            appPath = appPath.substr(0, found);
+        }
+        trpFilesPath = std::filesystem::path(appPath + "/game_data/" + title);
+
+        std::filesystem::path multiPath(trpFilesPath / "TrophyFiles");
+        if (fileList.size() > 1) {
+            multiPath = trpFilesPath / "TrophyFiles" / multiTrp;
+        }
+        std::filesystem::create_directories(multiPath / "Icons");
+
+        for (int i = 0; i < header.entry_num; i++) {
+            file.Seek(seekPos);
+            seekPos += (s64)header.entry_size;
+            trp_entry entry;
+            file.ReadRaw<u8>(&entry, sizeof(trp_entry));
```

Comment:
```suggestion
            file.Read(entry);
```
ReadRaw is intended for reading to raw pointers. Reading structures can be simplified